src/EnergyPlus/Pipes.cc

```diff
@@ -317,6 +237,9 @@ namespace Pipes {
 		NumSteamPipes = GetNumObjectsFound( "Pipe:Adiabatic:Steam" );
 		NumLocalPipes = NumWaterPipes + NumSteamPipes;
 		LocalPipe.allocate( NumLocalPipes );
+		for ( int p = 1; p <= NumLocalPipes; ++p ) {
+			LocalPipe( p ) = std::shared_ptr< LocalPipeData >( new LocalPipeData() );
```

Comment:
Should use `std::make_shared` instead. Or better yet, just use:

```cpp
for ( int p = 1; p <= NumLocalPipes; ++p ) {
  LocalPipe.emplace_back( std::make_shared< LocalPipeData  >() );
}
```

Although i'm a little surprised `LocalPipe.allocate( NumLocalPipes );` doesn't default construct the objects...